components/camel-minio/src/main/java/org/apache/camel/component/minio/MinioEndpoint.java

```diff
@@ -0,0 +1,294 @@
+        this.configuration = configuration;
+    }
+
+    public MinioClient getMinioClient() {
+        return minioClient;
+    }
+
+    public void setMinioClient(MinioClient minioClient) {
+        this.minioClient = minioClient;
+    }
+
+    public int getMaxMessagesPerPoll() {
+        return maxMessagesPerPoll;
+    }
+
+    /**
+     * Gets the maximum number of messages as a limit to poll at each polling.
+     * <p/>
+     * Gets the maximum number of messages as a limit to poll at each polling.
+     * The default value is 10. Use 0 or a negative number to set it as
+     * unlimited.
+     */
+    public void setMaxMessagesPerPoll(int maxMessagesPerPoll) {
+        this.maxMessagesPerPoll = maxMessagesPerPoll;
+    }
+
+    public int getMaxConnections() {
+        return maxConnections;
+    }
+
+    /**
+     * Set the maxConnections parameter in the minio client configuration
+     */
+    public void setMaxConnections(int maxConnections) {
+        this.maxConnections = maxConnections;
+    }
+
+    private String readInputStream(InputStream minioObject) throws IOException {
+        StringBuilder textBuilder = new StringBuilder();
+        try (Reader reader = new BufferedReader(new InputStreamReader(minioObject, Charset.forName(StandardCharsets.UTF_8.name())))) {
+            int c;
+            while ((c = reader.read()) != -1) {
+                textBuilder.append((char) c);
+            }
+        }
+        return textBuilder.toString();
+    }
+
+    private boolean bucketExists(String bucketName) throws Exception {
+        return minioClient.bucketExists(BucketExistsArgs.builder().bucket(bucketName).build());
+    }
+
+    private void makeBucket(String bucketName) throws Exception {
+        MakeBucketArgs.Builder makeBucketRequest = MakeBucketArgs.builder().bucket(bucketName).objectLock(getConfiguration().isObjectLock());
+        if (getConfiguration().getRegion() != null) {
+            makeBucketRequest.region(getConfiguration().getRegion());
+        }
+        minioClient.makeBucket(makeBucketRequest.build());
+    }
+
+    private void setBucketPolicy(String bucketName) throws Exception {
+        LOG.trace("Updating bucket {} with policy...", bucketName);
+        minioClient.setBucketPolicy(
+                SetBucketPolicyArgs.builder().bucket(bucketName).config(getConfiguration().getPolicy()).build());
+        LOG.trace("Bucket policy updated");
+    }
+
+    private void getObjectStat(String objectName, Message message) throws Exception {
+
+        String bucketName = getConfiguration().getBucketName();
+        StatObjectArgs.Builder statObjectRequest = StatObjectArgs.builder().bucket(bucketName).object(objectName);
+
+        if (getConfiguration().getServerSideEncryptionCustomerKey() != null) {
+            statObjectRequest.ssec(getConfiguration().getServerSideEncryptionCustomerKey());
+        }
+        if (getConfiguration().getOffset() != 0) {
```

Comment:
These checks are pretty redundant. Perhaps what we can do here, wrap these into centralized lambdas somewhere and just pass the function reference. Example:
```
private void checkMatchTagConfig(final MinioConfiguration configuration, final java.util.function.Consumer<String> fn) {
        if ( ObjectHelper.isNotEmpty(configuration.getMatchETag()) ) {
            fn.accept(configuration.getMatchETag());
        }
    }
```  
And then somewhere in the code:
```
checkMatchTagConfig(getConfiguration(), statObjectRequest::matchETag);
```